quinn/src/endpoint.rs

```diff
@@ -805,3 +855,16 @@ impl std::ops::Deref for EndpointRef {
         &self.0
     }
 }
+
+struct ReceiveContext<'a, 'b: 'c, 'c> {
+    cx: &'c mut Context<'b>,
+    now: Instant,
+    iovs: &'a mut [IoSliceMut<'a>; BATCH_SIZE],
+    metas: &'c mut [RecvMeta; BATCH_SIZE],
+}
+
+struct PollResult {
```

Comment:
Please derive `Default` on this, which should do the right thing to avoid initializing this manually.

Reply:
Nice.  I still find the explicit initialization more readable in the return statements of `poll_socket()`.  It is okay to only use `default()` in `drive_recv()`?

Reply:
Sure, that's what I had in mind, too!

Reply:
Sure, I pushed the updated version.  It's a bit unfortunate that I learn rust while I work on quinn.

Reply:
No problem, we're also just pretty picky, thanks for sticking with it!